lib/runner.js

```diff
@@ -61,6 +62,10 @@ optionChain(chainableMethods, function (opts, args) {
 	var fn;
 	var macroArgIndex;
 
+	if (this.hasBegunRunning) {
+		throw new Error('Cannot have nested tests or hooks');
```

Comment:
This could also happen with:

```js
test(async t => {
  // stuff
});

setImmediate(() => {
  test(t => {
    // not allowed!
  });
});
```

The message should include something about declaring tests async.

> All tests and hooks must be declared synchronously in your test file, and cannot be nested within other tests.